collect_app/src/main/java/org/odk/collect/android/preferences/dialogs/ResetDialogPreferenceFragmentCompat.java

```diff
@@ -143,10 +143,11 @@ protected void onPostExecute(List<Integer> failedResetActions) {
     }
 
     private void showProgressDialog() {
-        progressDialog = ProgressDialog.show(context,
-                        context.getString(R.string.please_wait),
-                        context.getString(R.string.reset_in_progress),
-                        true);
+        progressDialog = new ProgressDialogFragment();
```

Comment:
Keeping references to Fragments can be dangerous/flakey - if the host Activity is recreated, the Fragments will be as well, and we might end up with a reference to one of multiple fragments or no reference at all when we come to call `hideProgressDialog`.

We should switch to using `DialogFragmentUtils.showIfNotShowing`/`DialogFragmentUtils.dismiss` here for managing the dialog. That might mean pulling out a new subclass (see `FormLoadingDialogFragment`) or we could add a new overload to `DialogFragmentUtils` that lets us pass a tag in (as it currently always uses the class to calculate the tag).

Reply:
It was my firs solution but I ended up with some test failing on Firebase that I wasn't able to reproduce locally. Surprisingly this solution doesn't cause problems. I will give it another try maybe I missed something obvious since yesterday I also felt under the weather after getting my booster dose.

Reply:
Ok I was able to make it work so looks like I was affected by booster does indeed :smile: